assets/js/blocks/product-search/index.tsx

```diff
@@ -95,18 +109,102 @@ registerBlockVariation( 'core/search', {
 			/>
 		),
 	},
-	// @ts-ignore waiting for @types/wordpress__blocks update
-	isActive: ( blockAttributes, variationAttributes ) => {
-		return (
-			blockAttributes.query?.post_type ===
-			variationAttributes.query.postType
-		);
-	},
 	category: 'woocommerce',
 	keywords: [ __( 'WooCommerce', 'woo-gutenberg-products-block' ) ],
 	description: __(
 		'A search box to allow customers to search for products by keyword.',
 		'woo-gutenberg-products-block'
 	),
-	attributes: PRODUCT_SEARCH_ATTRIBUTES,
+	supports: {
+		align: [ 'wide', 'full' ],
+		inserter: isBlockVariationAvailable ? false : true,
+	},
+	example: {
+		attributes: {
+			hasLabel: true,
+		},
+	},
+	attributes,
+	transforms: isBlockVariationAvailable
+		? {
+				to: [
+					{
+						type: 'block',
+						blocks: [ 'core/search' ],
+						transform: () => {
+							return createBlock(
+								'core/search',
+								PRODUCT_SEARCH_ATTRIBUTES
+							);
+						},
+					},
+				],
+		  }
+		: {
+				from: [
+					{
+						type: 'block',
+						blocks: [ 'core/legacy-widget' ],
```

Comment:
Did you test how it works when transforming the widget into the new block, now? Does it require two steps?

Reply:
Sounds good! Yes, I'm not saying we need to do anything special, but I think we need to test that usecase just in case there is something weird, because I think the way it is it will require transforming the block twice.